services/QuillLMS/app/views/teacher_notifications/rollup_mailer/rollup.html.erb

```diff
@@ -0,0 +1,17 @@
+<p>Hi <%= @user.first_name %>,</p>
+
+<p>Here’s what happened in the last <%= @frequency %> on Quill:</p>
+
+<% @teacher_notifications.each do |classroom_name, notifications_by_type| %>
+  <h1><%= classroom_name %></h1>
+  <% notifications_by_type.each do |notification_type, notifications| %>
+    <% if notification_type == "TeacherNotifications::StudentCompletedDiagnostic" %>
```

Comment:
Since there's likely to be more of these in the future, I'd recommend using case statement for the partial:
```
partial = case notification_type
when "TeacherNotifications::StudentCompletedDiagnostic"
  :student_completed_diagnostic
when "TeacherNotifications::StudentCompletedAllDiagnosticRecommendations 
  :student_completed_all_diagnostic_recommendations
...
end

render partial, notifications: notifications
...

Reply:
Instead of the case statement you could also do:
```
partial = notification_type.constantize.name.demodulize.underscore
render partial, notification: notification
```

Reply:
I agree with Brendan's comment about minimizing logic in views. But to go a step further: you could abstract the if/case statement into a data structure, so you could then do 


```
notifications_by_type.each do |notification_type, notifications|
  <%= render LOOKUP_STRUCT[notification_type] %>
end
```

Reply:
I like @happythenewsad 's suggestion of a lookup table, so I've gone with that approach.

Reply:
I think @happythenewsad 's suggestion was the best one.

I'm realizing that while `notification_type.constantize.name.demodulize.underscore` is shorter it makes the code harder to follow when things go wrong.